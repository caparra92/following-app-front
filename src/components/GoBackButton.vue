<template>
    <div>
        <button @click="$router.go(-1)">
          <ion-icon size = "large" :icon="chevronBackOutline"></ion-icon>
        </button>
    </div>
</template>
<script setup lang="ts">
import { IonIcon } from '@ionic/vue';
import { chevronBackOutline } from 'ionicons/icons';
</script>
<style scoped>
button {
  display: flex;
  justify-content: center;
  align-items: center;
  background: transparent;
  width: 44px;
  border-radius: 5px;
}

ion-icon{
  color: var(--ion-color-dark);
}

</style>