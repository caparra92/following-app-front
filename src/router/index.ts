import { useStore } from "../stores/store";
import { createRouter, createWebHistory } from '@ionic/vue-router';
import { RouteRecordRaw } from 'vue-router';
import RegisterUserPage from '../views/RegisterUserPage.vue'
import LoginPage from '../views/LoginPage.vue';
import DashboardPage from '../views/DashboardPage.vue';
import ActivitiesPage from '../views/ActivitiesPage.vue';

const loggedIn = (to: any, from: any, next: any) => {
  const store = useStore();
  if(store.loggedIn) {
    next()
    return
  }
  next('/')
}

const routes: Array<RouteRecordRaw> = [
  {
    path: '/',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'Login',
    component: LoginPage
  },
  {
    path: '/register',
    name: 'Register',
    component: RegisterUserPage
  },
  {
    path: '/dashboard',
    name: 'Dashboard',
    component: DashboardPage,
    beforeEnter: loggedIn,
    meta: {
      requiresAuth: true,
    },
    children: [ 
      { 
        path: '/dashboard/activities', 
        name: 'Activities',
        component: ActivitiesPage,
        beforeEnter: loggedIn,
        meta: {
          requiresAuth: true
        }
      }
    ]
  }
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
});

router.beforeEach((to, from, next) => {
  const store = useStore();
  if (to.name !== 'Login' && !loggedIn) next({ name: 'Login' })
  else next()
});

export default router
