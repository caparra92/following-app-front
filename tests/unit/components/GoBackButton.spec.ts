import { mount, VueWrapper } from "@vue/test-utils";
import GoBackButton from "@/components/GoBackButton.vue";
import { createTestingPinia } from '@pinia/testing';
import router from "@/router";

describe('GoBackButton tests', () => {

    let wrapper: VueWrapper;

    beforeEach(() => {
        const mockRoute = {
            path: ''
          }

        const mockRouter = {
            go: jest.fn()
        }
        wrapper = mount(GoBackButton, {
            global: {
                plugins: [createTestingPinia(), router],
                mocks: {
                    $route: mockRoute,
                    $router: mockRouter
                  }
              },
        });
    });

    test('Should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Should verify that component exists', () => {
        expect(wrapper.exists()).toBeTruthy();
    });

    test('Should emit event when clicking a button', () => {
        const button = wrapper.find('button');
        button.trigger('click');        
        expect(wrapper.emitted('click')?.length).toBe(1);
    });
});