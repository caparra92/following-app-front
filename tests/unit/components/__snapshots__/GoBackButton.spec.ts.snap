// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoBackButton tests Should match with snapshot 1`] = `
<div><button>
    <ion-icon></ion-icon>
  </button></div>
`;
